File: src/main/java/ie/daithi/yeelight/enumeration/Event.java
package ie.daithi.yeelight.enumeration;

import lombok.Getter;

public enum Event {

    PLAY("media.play"), STOP("media.stop"), PAUSE("media.pause");

    @Getter
    private String value;

    Event (String value) {
        this.value = value;
    }

    public static Event fromValue(String value) {
		for(Event e : values()){
			if( e.getValue().equals(value)){
				return e;
			}
		}
		return null;
    }
}

File: src/main/kotlin/ie/daithi/yeelight/enumeration/Event.kt
package ie.daithi.yeelight.enumeration

enum class Event(private val value: String) {
    PLAY("media.play"), RESUME("media.resume"), STOP("media.stop"), PAUSE("media.pause");

    companion object {
        fun fromValue(value: String?): Event? {
            for (e in values()) {
                if (e.value.equals(value)) {
                    return e
                }
            }
            return null
        }
    }

}